cosmwasm/packages/enclave-ffi-types/src/types.rs

```diff
@@ -105,86 +81,48 @@ pub enum EnclaveError {
     EncryptionError,
     DecryptionError,
     NotImplemented,
+    Panic,
     /// Unexpected Error happened, no more details available
     Unknown,
-    Panic,
 }
 
+/// This type holds a pointer to a VmError that is boxed on the untrusted side
 #[repr(C)]
 #[derive(Debug, Display)]
-pub enum CryptoError {
-    /// The `DerivingKeyError` error.
-    ///
-    /// This error means that the ECDH process failed.
-    DerivingKeyError,
-    // {
-    //     self_key: [u8; 64],
-    //     other_key: [u8; 64],
-    // },
-    /// The `MissingKeyError` error.
-    ///
-    /// This error means that a key was missing.
-    MissingKeyError,
-    //  {
-    //     key_type: &'static str,
-    // },
-    /// The `DecryptionError` error.
-    ///
-    /// This error means that the symmetric decryption has failed for some reason.
-    DecryptionError,
-    /// The `ImproperEncryption` error.
-    ///
-    /// This error means that the ciphertext provided was imporper.
-    /// e.g. MAC wasn't valid, missing IV etc.
-    ImproperEncryption,
-    /// The `EncryptionError` error.
-    ///
-    /// This error means that the symmetric encryption has failed for some reason.
-    EncryptionError,
-    /// The `SigningError` error.
-    ///
-    /// This error means that the signing process has failed for some reason.
-    SigningError,
-    // {
-    //     hashed_msg: [u8; 32],
-    // },
-    /// The `ParsingError` error.
-    ///
-    /// This error means that the signature couldn't be parsed correctly.
-    ParsingError,
-    //  {
-    //     sig: [u8; 65],
-    // },
-    /// The `RecoveryError` error.
-    ///
-    /// This error means that the public key can't be recovered from that message & signature.
-    RecoveryError,
-    //  {
-    //     sig: [u8; 65],
-    // },
-    /// The `KeyError` error.
-    ///
-    /// This error means that a key wasn't vaild.
-    /// e.g. PrivateKey, PubliKey, SharedSecret.
-    // #[cfg(feature = "asymmetric")]
-    KeyError,
-    //  {
-    //     key_type: &'static str,
-    //     err: Option<secp256k1::Error>,
-    // },
-    // #[cfg(not(feature = "asymmetric"))]
-    // KeyError { key_type: &'static str, err: Option<()> },
-    // /// The `RandomError` error.
-    // ///
-    // /// This error means that the random function had failed generating randomness.
-    // #[cfg(feature = "std")]
-    // RandomError {
-    //     err: rand::Error,
-    // },
-    // #[cfg(feature = "sgx")]
-    RandomError, // {
-                 //     err: sgx_types::sgx_status_t,
-                 // }
+#[display(fmt = "VmError")]
```

Comment:
Cool. Didn't know #[display] could be used like this

Reply:
neither did I :smile: I had to look it up.

Reply:
It is specific to the `derive-more` crate though, so it's not standard.